### ISSUE TITLE
Methods set and update are not applied synchronously

### ISSUE THREAD

**Issue by many2023** [josdejong/svelte-jsoneditor#236]
```js
const options = {
        mode: "tree",
        mainMenuBar: true,
        navigationBar: false,
        statusBar: false,
        enableSort: false,
        enableTransform: false,
        colorPicker:false,
        readOnly: true,
        parser: LosslessJSONParser,
        pathParser: LosslessJSONParser,
        validationParser: LosslessJSONParser,
        onRenderMenu(items, context) {
            return items.slice(4)
        }
    }
    const editor = new JSONEditor({target: container, props: options})

    editor.set(json) // or editor.update(json)

    editor.expand(path => true)
```

**Comment by josdejong** [comment]
Can it be that you're using `jsoneditor` instead of `svelte-jsoneditor`?

**Comment by many2023** [comment]
absolutely svelte-jsoneditor，Because LosslessJSON is worked

**Comment by josdejong** [comment]
Ok clear, I thought that because you're using options like `enableSort` and `enableTransform` which are not available in `svelte-jsoneditor`. On a side note, I do not think you should configure `pathParser` and `validationParser` to use the lossless JSON parser. 

I see the editor does not synchronously update it's contents, and the editor is not yet updated right after `editor.set`. If you expand after a timeout of 0 things work as expected:

```js
editor.set(json) // or editor.update(json)

setTimeout(() => editor.expand(path => true))
```
What also works is passing the contents directly at creation:

```js
const options = {
  // ...
  content: { json }
}
const editor = new JSONEditor({target: container, props: options})
```

I'll look at a solution to this. Maybe it is necessary to make `set` and `update` async to prevent issues like this.

**Comment by josdejong** [comment]
Fixed now in `v0.16.0` by making these methods async, making sure their effects are applied and the editor is fully re-rendered. So you can do:

```js
await editor.set(json)
await editor.expand(path => true)
```
